pkg/resources/common/formatter.go

```diff
@@ -184,3 +220,27 @@ func excludeValues(request *types.APIRequest, unstr *unstructured.Unstructured)
 		}
 	}
 }
+
+func getNamespaceFromResource(obj types.APIObject) string {
+	unstr, ok := obj.Object.(*unstructured.Unstructured)
+	if !ok {
+		return ""
+	}
+
+	// If we have a backingNamespace, use that
+	if statusRaw, ok := unstr.Object["status"]; ok {
+		if statusMap, ok := statusRaw.(map[string]interface{}); ok {
+			if backingNamespace, ok := statusMap["backingNamespace"].(string); ok && backingNamespace != "" {
+				return backingNamespace
+			}
+		}
+	}
+
+	// Otherwise, if the id has a slash, we will interpret that
+	parts := strings.SplitN(obj.ID, "/", 2)
+	if len(parts) == 2 {
+		return parts[0] + "-" + parts[1]
+	}
+
+	return obj.ID
```

Comment:
We are returning obj.ID. How is obj.ID a namespace ?

Reply:
Updated code to Eric's recommendation and added a comment to explain.